typescript/packages/subsurface-viewer/src/storybook/layers/WellsLayer.stories.tsx

```diff
@@ -769,6 +776,117 @@ export const ReducedWellNameClutter2D: StoryObj<
+                wellNameAtTop: true,
+                wellHeadStyle: { size: 4 },
+                hideOverlappingWellNames: true,
+                outline: true,
+                ZIncreasingDownwards: false,
+                // eslint-disable-next-line
+                // @ts-ignore
+                dataTransform: (data_in) => {
+                    // Simplify well paths by reducing number of segments/nodes.
+                    const data =
+                        data_in as FeatureCollection<GeometryCollection>;
+
+                    const no_wells = data.features.length;
+                    for (let well_no = 0; well_no < no_wells; well_no++) {
+                        const geometryCollection = data.features[well_no]
+                            .geometry as GeometryCollection;
+                        const lineString = geometryCollection
+                            ?.geometries[1] as LineString;
+
+                        if (lineString.coordinates?.length === undefined) {
+                            continue;
+                        }
+
+                        const properties = data.features[well_no]
+                            .properties as GeoJsonProperties;
+                        if (properties) {
+                            const mds = properties["md"][0];
+                            const [newPoints, newMds] = simplify(
+                                lineString.coordinates as Position3D[],
+                                mds,
+                                props.coarseWellsToleranceFactor
+                            );
+                            lineString.coordinates = newPoints as Position3D[];
+
+                            properties["md"][0] = newMds;
+                        }
+                    }
+
+                    return data;
+                },
+            }),
+            new AxesLayer({
+                id: "axes-layer",
+                bounds: [450000, 6781000, 0, 464000, 6791000, 3500],
+            }),
+        ],
+        bounds: [450000, 6781000, 464000, 6791000] as BoundingBox2D,
+        views: {
+            layout: [1, 1] as [number, number],
+            viewports: [
+                {
+                    id: "view_1",
+                    show3D: true,
+                },
+            ],
+        },
+    };
+
+    return (
+        <SubsurfaceViewer {...propsWithLayers}>
+            {
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                /* @ts-expect-error */
+                <View id="view_1">
+                    <ViewFooter>
+                        Coarsen wells using &quot;dataTransform&quot; property.
+                        Adjustable factor typical range: [0.001, 1000]
+                    </ViewFooter>
+                </View>
+            }
+        </SubsurfaceViewer>
+    );
+};
+
+export const CoarseWellFactor: StoryObj<typeof CoarseWellFactorComponent> = {
+    args: {
+        coarseWellsToleranceFactor: 0.01,
```

Comment:
Suggestion: use `argTypes` to set this control to a slider

Reply:
Made into slider.